lib/charms/opensearch/v0/opensearch_locking.py

```diff
@@ -291,35 +304,36 @@ def acquired(self) -> bool:  # noqa: C901
                     # https://www.elastic.co/guide/en/elasticsearch/reference/8.13/docs-index_.html#index-wait-for-active-shards
                     if response["_shards"]["failed"] > 0:
-                        logger.error("Failed to write OpenSearch lock document to all nodes")
+                        logger.error("Failed to write OpenSearch lock document to all nodes.")
+                        logger.debug(
+                            "[Node lock] Deleting OpenSearch lock after failing to write to all nodes"
+                        )
+                        # Delete document id 0
+                        self._opensearch.request(
+                            "DELETE",
+                            endpoint=f"/{self.OPENSEARCH_INDEX}/_doc/0?refresh=true",
+                            host=host,
+                            alt_hosts=alt_hosts,
+                            retries=10,
+                        )
+                        logger.debug(
+                            "[Node lock] Deleted OpenSearch lock after failing to write to all nodes"
+                        )
                         return False
-            try:
-                unit = self._unit_with_lock(host)
-            except OpenSearchHttpError:
-                logger.exception("Error checking which unit has OpenSearch lock")
-                return False
+                    # This unit has OpenSearch lock
+                    unit = self._charm.unit.name
             if unit == self._charm.unit.name:
                 # Lock acquired
                 # Release peer databag lock, if any
                 logger.debug("[Node lock] Acquired via opensearch")
                 self._peer.release()
                 logger.debug("[Node lock] Released redundant peer lock (if held)")
                 return True
-            if unit or online_nodes >= 2:
+            if unit:
                 # Another unit has lock
-                # (Or document deleted after request to create document & before request in
-                # `self._unit_with_lock()`)
                 logger.debug(f"[Node lock] Not acquired. Unit with opensearch lock: {unit}")
                 return False
-            # If online_nodes == 1, we should acquire the lock via the peer databag.
-            # If we acquired the lock via OpenSearch and this unit was stopping, we would be unable
-            # to release the OpenSearch lock. For example, when scaling to 0.
-            # Then, when 1+ OpenSearch nodes are online, a unit that no longer exists could hold
-            # the lock.
-            # Note: if online_nodes > 1, this situation is still possible (e.g. if this unit was
-            # stopping and another unit went offline simultaneously)—but it's an edge case we don't
-            # support (to reduce complexity & improve robustness in other cases).
-            # If online_nodes > 1, we should re-attempt to acquire the OpenSearch lock.
+            assert online_nodes == 1
```

Comment:
What is the point of this statement

Reply:
My concern here is that, not using the `OpenSearchError` and `AssertionError` instead. That will cause the charm to go into "ERROR" state as we never catch the later